core/src/main/java/com/linecorp/armeria/client/Http2ResponseDecoder.java

```diff
@@ -245,11 +247,18 @@ public int onDataRead(
         }
 
         final long maxContentLength = res.maxContentLength();
-        if (maxContentLength > 0 && res.writtenBytes() > maxContentLength - dataLength) {
-            res.close(ContentTooLargeException.get());
-            throw connectionError(INTERNAL_ERROR,
-                                  "content length too large: %d + %d > %d (stream: %d)",
-                                  res.writtenBytes(), dataLength, maxContentLength, streamId);
+        final long writtenBytes = res.writtenBytes();
+        if (maxContentLength > 0 && writtenBytes > maxContentLength - dataLength) {
+            final long transferred = LongMath.saturatedAdd(writtenBytes, dataLength);
+            res.close(ContentTooLargeException.builder()
+                                              .maxContentLength(maxContentLength)
+                                              .contentLength(res.headers())
+                                              .transferred(transferred)
+                                              .build());
+            throw connectionError(
+                    INTERNAL_ERROR,
+                    "content too large: transferred(%d) + delta(%d) > limit(%d) (stream: %d)",
```

Comment:
Do we want to update this error message accordingly? 
```suggestion
                    "content too large: transferred(%d) + contentLength(%d) > maxContentLength(%d) (stream: %d)",
```